Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -732,13 +735,43 @@ def sandbox_report(md5, details):
     return response
 
 
+def login_command():
+    auth = login()
+    jsession_id = auth[:auth.index(';')]
+    DEFAULT_HEADERS['cookie'] = jsession_id
+    readable_output = '### Created Zscaler Session\nSession ID: {}'.format(jsession_id)
+    return CommandResults(
+        outputs_prefix='Zscaler.SessionID',
+        outputs=jsession_id,
+        readable_output=readable_output,
+        raw_response=auth)
+
+
+def logout_command():
+    logout()
+    return "API session logged out of Zscaler successfully."
+
+
+def activate_command():
+    activate_changes()
+    return "Changes have been activated successfully."
+
+
 ''' EXECUTION CODE '''
 
 
 def main():
-    auth = login()
-    jsession_id = auth[:auth.index(';')]
-    DEFAULT_HEADERS['cookie'] = jsession_id
+    params = demisto.params()
+    args = demisto.args()
+    auto_login = params.get('auto_login')
+    if args.get('session_id'):
+        DEFAULT_HEADERS['cookie'] = args['session_id']
+    elif auto_login or demisto.command() in ('test-module', 'zscaler-login'):
+        auth = login()
+        DEFAULT_HEADERS['cookie'] = auth[:auth.index(';')]
+    else:
+        return_error("Error: session_id is missing. When not using Auto Login, you must provide a session_id. "
+                     "Please run zscaler-login to get a valid session id or turn Auto Login on.")
```

Comment:
putting this block here will mean e.g that the `login()` will not be handled in the try/except block. any reason for this? or can we move to the try/except block?

Reply:
This is in line with the original implementation, and I see no reason to change it.
Moving it inside the existing `try/except/finally` block might add a redundant logout attempt and a redundant activate attempt, while not adding more log info, as the error will be printed to the `error` channel since it's raised back.